types/reach__menu-button/index.d.ts

```diff
@@ -45,14 +46,14 @@ export type ResolvedMenuLinkProps<T> = T extends keyof JSX.IntrinsicElements
   ? JSX.IntrinsicElements[T]
   : T;
 
-export type ResolvedMenuLinkComponent<T> = T extends keyof JSX.IntrinsicElements
-  ? T
-  : React.ComponentType<T>;
+export type ResolvedMenuLinkComponent<T> =
+  | keyof JSX.IntrinsicElements
+  | React.ComponentType<T>;
```

Comment:
Why did this get switched to a union?

Reply:
Because `T extends keyof JSX.IntrinsicElements` does not work and I can't figured the issue without the union.

If `T` is a React component named `Link`, `T extends keyof JSX.IntrinsicElements` returns true for some reason...